lib/mutant/isolation.rb

```diff
@@ -1,31 +1,35 @@
 module Mutant
   # Module providing isolation
-  module Isolation
+  class Isolation
     Error = Class.new(RuntimeError)
 
-    module None
-
+    class None < self
       # Call block in isolation
       #
       # @return [Object]
       #
       # @raise [Error]
       #   if block terminates abnormal
-      def self.call(&block)
-        block.call
+      def call
+        yield
       rescue => exception
```

Comment:
Right now I've no counter example where I wanted to capture an exception that does NOT inherit from standard error. 

This iteration is only to get this into a verifiable state. I'll do a followup that does a much better job for propagating the exceptions captured in the child, I'll revisit your point here.